### ISSUE TITLE
Auto-versioning bug

### ISSUE THREAD

**Issue by gaynetdinov** [boldpoker/edeliver#95]
Hello.
`edeliver` version is 1.2.8.

I have the following autoversioning setting in `.deliver/config`:
```
AUTO_VERSION=git-branch+git-revision+mix-env
```

I just ran 
```
$ mix edeliver build release --mix-env=qa2 --branch=my-feature --verbose 
```
and got build called 
```
my_app-0.0.2+add-demo-env-698f0ad-qa2.tar.gz
```
Where `add-demo-env` is one of the branches(the first one in `git branch`) and `698f0ad` is commit in requested branch called `my-feature`. So the build got built using commit from target branch, but with completely different branch name. I have no idea how to debug it, I ran the command multiple times and every time I got the wrong name of the release.

**Comment by gaynetdinov** [comment]
Ah, I went to build host and removed `add-demo-env` branch there locally, after that build got built with correct branch(which was requested using `--branch` option). Maybe that helps you to debug this issue.

**Comment by bharendt** [comment]
Was the `698f0ad` commit in both branches? edeliver tries to detect the branch which contains the commit by using `git branch --contains <revision>`. I think this is not ideal, at least if a branch was first built (and because of that remains on the build host) and later merged (and built again). Then there would be two branches on the build host containing that commit. So maybe we should just pass the branch from the `--branch` option to the [Release.Version](https://github.com/boldpoker/edeliver/blob/master/lib/mix/tasks/release_version.ex) mix task. I think it might be the same or a similar problem in [that issue](https://github.com/boldpoker/edeliver/issues/74).

**Comment by gaynetdinov** [comment]
Yes, `698f0ad` is in both branches.

**Comment by bharendt** [comment]
Thanks. I think this commit should fix that.

**Comment by bharendt** [comment]
This fix is included in edeliver version [`1.2.9`](https://hex.pm/packages/edeliver/1.2.9).
